thunder/tests/test_networks.py

```diff
@@ -359,3 +359,56 @@ def test_quantization():
     assert len(sd) == len(sd2)
     for k, v in sd.items():
         assert_close(v, sd2[k])
+
+
+@thunder.tests.framework.requiresCUDA
+@pytest.mark.skip(reason="Issue 1240")
+def test_thunderfx_mistral_nemo_small():
+    """
+    Runs a small version of Mistral-NeMo
+
+    This is largely based on code from Alexandros Koumparoulis.
+    """
+    import transformers
+
+    model_id = "mistralai/Mistral-Nemo-Base-2407"
+
+    # Setup a "small" version of NeMo-Mistral that does not require downloading
+    # weights. This is not a configuration that is worth benchmarking.
+    # This was created by using
+    #   MistralConfig(num_hidden_layers=1, max_position_embeddings=1024)
+    # and then manually diffing that returned object with:
+    #   transformers.AutoConfig.from_pretrained(model_id)
+    # until they lined up sans the hidden and embeddings changes, above.
+    config = transformers.models.mistral.configuration_mistral.MistralConfig(
+        num_hidden_layers=1,
+        torch_dtype=torch.bfloat16,
+        max_position_embeddings=1024,
+        architectures=["MistralForCausalLM"],
+        hidden_size=5120,
+        rms_norm_eps=1e-05,
+        rope_theta=1000000.0,
+        sliding_window=None,
+        vocab_size=131072,
+        head_dim=128,
+        _name_or_path=model_id,
+    )
+    model = transformers.AutoModelForCausalLM.from_config(config, trust_remote_code=False)
+    device = torch.device("cuda")
+    model.to(device)
+    model.train()
+    th_backend = thunder.dynamo.ThunderCompiler()
+    mdl = torch.compile(model, backend=th_backend)
+    del model
```

Comment:
Curious to know if this is actually needed. As per my understanding, `mdl` will use and hold onto the references for the parameters of `model`, so `del model` shouldn't reclaim a lot of memory except for clearing the `nn.Module` Python object.

Thanks!

Reply:
Indeed, not needed. I just did this as a technique for myself to make sure I didn't use `model` after compiling it; with the `del` were I to screw up and accidentally use the *uncompiled* model, Python would die with an unknown variable error; without it the script would just silently test the wrong thing.

I am happy to delete it for clarity. Since I have to re-upload to unskip, I'll fix it as part of that.

Reply:
Got it, thanks for clarifying